api/v1alpha1/externaldns_types.go

```diff
@@ -408,6 +408,21 @@ type ExternalDNSSource struct {
 	// +kubebuilder:default:=Ignore
 	// +required
 	HostnameAnnotationPolicy HostnameAnnotationPolicy `json:"hostnameAnnotation"`
+
+	// FQDNTemplate sets a templated string that's used to generate DNS names
+	// from sources that don't define a hostname themselves.
+	// Accepts comma separated list for multiple global FQDN.
+	//
+	// Especially useful when HostnameAnnotationPolicy is set to Ignore.
+	//
+	// Provided template should follow the syntax defined for text/template Go package,
+	// see https://pkg.go.dev/text/template.
+	// Annotations inside the template correspond to the definition of the source data structure.
+	// Example: "{{.Name}}.example.com" would be expanded to "myservice.example.com" for service source
```

Comment:
We should document what the default (empty) value signifies.  Presumably that means no DNS record is created if the source resource doesn't specify a host name, right?

Reply:
Currently: default value (empty string) means no `--fqdn-template` flag will be added which in combination with the default hostname annotation policy (Ignore) would result into externalDNS container failing in the loop.

A possibility which I considered was add `--fdqn-template` even if the value is empty this would result in the behavior which you described: no records will be populated because none would match the empty FQDNTemplate. 

But I thought that it would be better to let the user know that the configuration is not optimal. That's why I added a task in [CFE-114](https://issues.redhat.com/browse/CFE-114) to add the validation of FQDNTemplate if the hostname annotation policy is Ignore.

Updated the doc to state explicitly that the value should not be empty if the hostname policy is Ignore.

Reply:
Is it important to allow an empty value?

By the way, as this allows multiple values, it should be a slice of strings, and we should document what specifying multiple strings does (presumably, for a given service, external-dns creates one DNS record for each string).

Reply:
Yes, true it should have been a slice, my bad. Specifying multiple strings would create 1 DNS record for every **matching** domain. And the domain is found from the list of zone IDs or from the domains filtered by the domain filter (in the future).